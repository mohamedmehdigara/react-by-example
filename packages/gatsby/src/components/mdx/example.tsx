import React, { FC, ReactNode, useCallback, useState } from 'react'
import styled from 'styled-components'
import { CodeSandboxIcon, ResetIcon } from '../icons'

const Render = styled.div`
  display: flex;
  padding: 1em;
  border: 1px solid var(--app-foreground);
  border-radius: var(--app-border-radius);
  align-items: center;
  justify-content: center;
`
const FlexRight = styled.div`
  display: flex;
  justify-content: flex-end;
`

const CodeSandboxLink = styled.a`
  display: flex;
`

const ResetButton = styled.button`
  display: flex;
  margin: 0;
  padding: 0;
  color: var(--app-foreground);
  background: none;
  border: none;
  cursor: pointer;

  :focus,
  :active {
    color: var(--app-cyan);
    outline: 2px solid var(--app-cyan);
  }
`

export interface ExampleProps {
  code?: ReactNode
  render?: ReactNode
  link?: string
}

export const Example: FC<ExampleProps> = ({ code, render, link }) => {
  const [resetKey, setResetKey] = useState(Date.now())
  const triggerReset = useCallback(() => {
    setResetKey(Date.now())
  }, [])
  return (
    <>
      {code}
      <Render key={resetKey}>
        <div>{render}</div>
      </Render>
      {link && (
        <FlexRight>
          <ResetButton
            type="button"
            title="reset example"
            onClick={triggerReset}
          >
            <ResetIcon />
          </ResetButton>
          <CodeSandboxLink
            target="_blank"
            rel="noreferrer"
            href={link}
            title="edit on CodeSandbox"
          >
            <CodeSandboxIcon />
          </CodeSandboxLink>
        </FlexRight>
      )}
    </>
  )
}
