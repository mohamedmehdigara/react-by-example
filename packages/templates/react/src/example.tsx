import React, { FC, ReactElement } from 'react'

export const Simple: FC = () => <h1>Hello world!</h1>

export default (): ReactElement => <Simple />
