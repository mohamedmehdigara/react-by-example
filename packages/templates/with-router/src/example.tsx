import React, { FC } from 'react'

export const Simple: FC = () => <h1>Hello world!</h1>

export default <Simple />
