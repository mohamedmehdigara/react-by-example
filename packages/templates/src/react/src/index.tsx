import React from 'react'
import ReactDOM from 'react-dom'
import example from './example'

ReactDOM.render(<>{example}</>, document.getElementById('root'))
