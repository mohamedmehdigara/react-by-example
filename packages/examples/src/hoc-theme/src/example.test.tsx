import { cleanup, render, RenderResult } from '@testing-library/react'
import React from 'react'
import example, { ThemedCard, ThemedParagraph } from './example'
import { DarkTheme, LightTheme } from './theme-library'

const stubConsole = () => {
  let spy: jest.SpyInstance

  beforeEach(() => {
    spy = jest.spyOn(console, 'log').mockImplementation(() => {})
  })

  afterEach(() => {
    spy.mockRestore()
  })
}

describe('ThemedParagraph', () => {
  let component: RenderResult

  stubConsole()

  afterEach(() => {
    cleanup()
  })

  describe('dark', () => {
    beforeEach(() => {
      component = render(
        <DarkTheme>
          <ThemedParagraph>Lorem ipsum dolor sit amet.</ThemedParagraph>
        </DarkTheme>,
      )
    })

    it('should snapshot', () => {
      expect(component.container).toMatchSnapshot()
    })
  })

  describe('light', () => {
    beforeEach(() => {
      component = render(
        <LightTheme>
          <ThemedParagraph>Lorem ipsum dolor sit amet.</ThemedParagraph>
        </LightTheme>,
      )
    })

    it('should snapshot', () => {
      expect(component.container).toMatchSnapshot()
    })
  })
})

describe('ThemedCard', () => {
  let component: RenderResult

  afterEach(() => {
    cleanup()
  })

  describe('dark', () => {
    beforeEach(() => {
      component = render(
        <DarkTheme>
          <ThemedCard>Lorem ipsum dolor sit amet.</ThemedCard>
        </DarkTheme>,
      )
    })

    it('should snapshot', () => {
      expect(component.container).toMatchSnapshot()
    })
  })

  describe('light', () => {
    beforeEach(() => {
      component = render(
        <LightTheme>
          <ThemedCard>Lorem ipsum dolor sit amet.</ThemedCard>
        </LightTheme>,
      )
    })

    it('should snapshot', () => {
      expect(component.container).toMatchSnapshot()
    })
  })
})

describe('example', () => {
  let component: RenderResult

  stubConsole()

  beforeEach(() => {
    component = render(example)
  })

  afterEach(() => {
    cleanup()
  })

  it('should snapshot', () => {
    expect(component.container).toMatchSnapshot()
  })
})
