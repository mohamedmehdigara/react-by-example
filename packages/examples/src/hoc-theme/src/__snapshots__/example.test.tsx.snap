// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedCard dark should snapshot 1`] = `
<div>
  <div
    style="background-color: rgb(40, 42, 54);"
  >
    Lorem ipsum dolor sit amet.
  </div>
</div>
`;

exports[`ThemedCard light should snapshot 1`] = `
<div>
  <div
    style="background-color: rgb(248, 248, 242);"
  >
    Lorem ipsum dolor sit amet.
  </div>
</div>
`;

exports[`ThemedParagraph dark should snapshot 1`] = `
<div>
  <p
    style="color: rgb(248, 248, 242);"
  >
    Lorem ipsum dolor sit amet.
  </p>
</div>
`;

exports[`ThemedParagraph light should snapshot 1`] = `
<div>
  <p
    style="color: rgb(40, 42, 54);"
  >
    Lorem ipsum dolor sit amet.
  </p>
</div>
`;

exports[`example should snapshot 1`] = `
<div>
  <div
    style="background-color: rgb(40, 42, 54);"
  >
    <p
      style="color: rgb(248, 248, 242);"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Soluta ipsum sapiente ut pariatur. Aliquam suscipit consequuntur similique repellat praesentium eos odit. Ad ipsum voluptatibus natus dignissimos rerum. Laborum, molestias temporibus?
    </p>
    <p
      style="color: rgb(248, 248, 242);"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Soluta ipsum sapiente ut pariatur. Aliquam suscipit consequuntur similique repellat praesentium eos odit. Ad ipsum voluptatibus natus dignissimos rerum. Laborum, molestias temporibus?
    </p>
  </div>
  <div
    style="background-color: rgb(248, 248, 242);"
  >
    <p
      style="color: rgb(40, 42, 54);"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Soluta ipsum sapiente ut pariatur. Aliquam suscipit consequuntur similique repellat praesentium eos odit. Ad ipsum voluptatibus natus dignissimos rerum. Laborum, molestias temporibus?
    </p>
    <p
      style="color: rgb(40, 42, 54);"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Soluta ipsum sapiente ut pariatur. Aliquam suscipit consequuntur similique repellat praesentium eos odit. Ad ipsum voluptatibus natus dignissimos rerum. Laborum, molestias temporibus?
    </p>
  </div>
</div>
`;
